cfme/tests/services/test_provision_stack.py

```diff
@@ -0,0 +1,110 @@
+import pytest
+
+from cfme.services.catalogs.catalog_item import CatalogItem
+from cfme.services.catalogs.catalog import Catalog
+from cfme.services.catalogs.orchestration_template import OrchestrationTemplate
+from cfme.services.catalogs.service_catalogs import ServiceCatalogs
+from cfme.services import requests
+from cfme.web_ui import flash
+from utils import testgen
+from utils.randomness import generate_random_string
+from utils.providers import setup_provider
+from utils.log import logger
+from utils.wait import wait_for
+
+
+pytestmark = [
+    pytest.mark.usefixtures("logged_in"),
+    pytest.mark.meta(server_roles="+automate"),
+    pytest.mark.ignore_stream("5.2", "5.3", "upstream")
+]
+
+
+def pytest_generate_tests(metafunc):
+    # Filter out providers without templates defined
+    argnames, argvalues, idlist = testgen.cloud_providers(metafunc, 'provisioning')
+    new_argvalues = []
+    new_idlist = []
+    for i, argvalue_tuple in enumerate(argvalues):
+        args = dict(zip(argnames, argvalue_tuple))
+        if not args['provisioning']:
+            # Don't know what type of instance to provision, move on
+            continue
+
+        # required keys should be a subset of the dict keys set
+        if not {'stack_provisioning'}.issubset(args['provisioning'].viewkeys()):
+            # Need image for image -> instance provisioning
+            continue
+
+        new_idlist.append(idlist[i])
+        new_argvalues.append([args[argname] for argname in argnames])
+
+    testgen.parametrize(metafunc, argnames, new_argvalues, ids=new_idlist, scope="module")
+
+
+@pytest.fixture
+def provider_init(provider_key):
+    """cfme/infrastructure/provider.py provider object."""
+    try:
+        setup_provider(provider_key)
+    except Exception:
+        pytest.skip("It's not possible to set up this provider, therefore skipping")
+
+
+@pytest.yield_fixture(scope="function")
+def dialog(provisioning):
+    dialog = "dialog_" + generate_random_string()
+    template_type = provisioning['stack_provisioning']['template_type']
+    template_name = provisioning['stack_provisioning']['template_name']
+    # template_type, template_name = map(provisioning.get('template_type', 'stack'))
+    orch_dialog = OrchestrationTemplate(template_type=template_type,
+                               template_name=template_name)
+    orch_dialog.create_service_dialog(dialog)
+    yield dialog
+
+
+@pytest.yield_fixture(scope="function")
+def catalog():
+    cat_name = "cat_" + generate_random_string()
+    catalog = Catalog(name=cat_name, description="my catalog")
+    catalog.create()
+    yield catalog
```

Comment:
Good question. It seems like this and the dialog fixture should probably have some cleanup code after the yield